### ISSUE TITLE
early stopping

### ISSUE THREAD

**Issue by amir-zeraati** [pytorch/pytorch#96555]
### 🚀 The feature, motivation and pitch

It would be nice and easy to have the early stopping function/feature in Pytorch so that neural network training can stop before overfitting based on some criteria for instance algorithm stops after 3 consequent epochs with very small changes in validation loss. At the moment this is possible in the training loop by adding a condition but if this is done by a separate specific function, it would be much easier for users to utilize it.
